docs/recipes/recipes.md

```diff
@@ -0,0 +1,428 @@
+# Recipes and Examples
+
+This page provides a collection of common use-cases and how to implement them using Slint.
+
+## Get Started
+
+### A clickable Button
+
+```slint
+import { VerticalBox, Button } from "std-widgets.slint";
+export Recipe := Window {
+    property <int> counter: 0;
+    VerticalBox {
+        button := Button {
+            text: "Button, pressed \{root.counter} times";
+            clicked => {
+                root.counter += 1;
+            }
+        }
+    }
+}
+```
+
+### React to a Button in native code
+
+```slint
+import { VerticalBox, Button } from "std-widgets.slint";
+export Recipe := Window {
+    property <int> counter: 0;
+    callback button_pressed <=> button.clicked;
+    VerticalBox {
+        button := Button {
+            text: "Button, pressed \{root.counter} times";
+        }
+    }
+}
+```
+
+<details data-snippet-language="rust">
+<summary>Rust code</summary>
+In Rust you can write
+
+```rust,no_run
+slint::slint!(import { Recipe } from "docs/recipes/button_native.slint";);
```

Comment:
if it is not run, the directory doesn't need to be valid. should probably just be `"button_native.slint"`.

Reply:
it is not previewed by the wasm in the browser.  But it is still tested by the doctest test that checks for slint compile error.